### PR TITLE
KarlsenHashV2-Fishhash Karlsen Node & CPU miner for Testnet-1

### PR DESCRIPTION
**Pull request by wam-rd**
First implementation of fishhash algo on the karlsend & karlsenminer.
This first iteration is as close as possible to fishhash whitepaper (same seed, same prebuilt dataset epoch 488, etc)
The kheavyhash kernel is still present in the codesource for future algo switch but not used yet
It means this version is KarlsenHashV2 compatible only

todo : 
- better comments on codesource 
- test suite for fishhash

algo iteration name : fishhash-kls-0.0.1

### CODE REVIEW

File: domain/consensus/utils/pow/fishhash.go
Comment (okilisan) [inline domain/consensus/utils/pow/fishhash.go:1]: Ok,as mentioned we have some hard coded value we will have to put in miner options. Ok for testnet I think.

File: domain/consensus/utils/pow/heavyhash_test.go
Comment (okilisan) [inline domain/consensus/utils/pow/heavyhash_test.go:1]: Not sure about this test change ?

**Review by okilisan — APPROVED** [review]
Some minor comments
Everything ok for me.

File: app/app.go [hunk 1]
```diff
@@ -82,6 +83,12 @@ func (app *karlsendApp) main(startedChan chan<- struct{}) error {
 
 	// Show version at startup.
 	log.Infof("Version %s", version.Version())
+	log.Infof("Using KarlsenHashV2 impl: %s", pow.GetHashingAlgoVersion())
+	if !app.cfg.Testnet && !app.cfg.Devnet && !app.cfg.Simnet {
```
Comment (lemois-1337) [inline app/app.go:87]: Ok for me and very good that builds for `testnet-1` are not allowed to connect to `mainnnet`.

File: domain/consensus/utils/pow/fishhash.go
Comment (lemois-1337) [reply domain/consensus/utils/pow/fishhash.go:1]: Yes, if needed in GPU-miner later in testnet we will create a special build for it as well.

File: domain/consensus/utils/pow/heavyhash_test.go
Comment (lemois-1337) [reply domain/consensus/utils/pow/heavyhash_test.go:1]: Its ok for now. unit tests didn't pass currently anyway. It is being fixed in parallel while `testnet-1` is running.

File: domain/dagconfig/params.go [hunk 2]
```diff
@@ -287,7 +287,7 @@ var TestnetParams = Params{
 	RPCPort:     "42210",
 	DefaultPort: "42211",
 	DNSSeeds: []string{
-		"testnet-1.karlsencoin.com",
+		"testnet-1-dnsseed.karlsencoin.com",
```
Comment (lemois-1337) [inline domain/dagconfig/params.go:290]: @okilisan They are not yet running, is it ok?

File: version/version.go [hunk 3]
```diff
@@ -9,8 +9,8 @@ import (
 const validCharacters = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz-"
 
 const (
-	appMajor uint = 1
-	appMinor uint = 1
+	appMajor uint = 2
```
Comment (lemois-1337) [inline version/version.go:12]: Ok, we will have a `mainnet` release with `1.2.0` in couple of days as well. Well aligned.

File: domain/dagconfig/params.go (hunk 2 above)
Comment (okilisan) [reply domain/dagconfig/params.go:290]: I'm setting it up. It should be ok

**Review by lemois-1337 — APPROVED** [review]

**Comment by lemois-1337** [comment]
Looks good, merged into `testnet-1` branch.

**Pull request merged**